src/libraries/System.Private.CoreLib/src/System/SearchValues/IndexOfAnyAsciiSearcher.cs

```diff
@@ -602,6 +676,416 @@ private static unsafe bool TryLastIndexOfAny<TNegator>(ref short searchSpace, in
+                    ref short twoVectorsAwayFromEnd = ref Unsafe.Add(ref searchSpace, searchSpaceLength - (2 * Vector256<short>.Count));
+
+                    do
+                    {
+                        Vector256<short> source0 = Vector256.LoadUnsafe(ref currentSearchSpace);
+                        Vector256<short> source1 = Vector256.LoadUnsafe(ref currentSearchSpace, (nuint)Vector256<short>.Count);
+
+                        Vector256<byte> result = IndexOfAnyLookup<TNegator, TOptimizations>(source0, source1, asciiBitmap256, secondBitmap256, offset256);
+                        if (result != Vector256<byte>.Zero)
+                        {
+                            return TResultMapper.FirstIndex<TNegator>(ref searchSpace, ref currentSearchSpace, result);
+                        }
+
+                        currentSearchSpace = ref Unsafe.Add(ref currentSearchSpace, 2 * Vector256<short>.Count);
+                    }
+                    while (Unsafe.IsAddressLessThan(ref currentSearchSpace, ref twoVectorsAwayFromEnd));
+                }
+
+                // We have 1-32 characters remaining. Process the first and last vector in the search space.
+                // They may overlap, but we'll handle that in the index calculation if we do get a match.
+                Debug.Assert(searchSpaceLength >= Vector256<short>.Count, "We expect that the input is long enough for us to load a whole vector.");
+                {
+                    ref short oneVectorAwayFromEnd = ref Unsafe.Add(ref searchSpace, searchSpaceLength - Vector256<short>.Count);
+
+                    ref short firstVector = ref Unsafe.IsAddressGreaterThan(ref currentSearchSpace, ref oneVectorAwayFromEnd)
+                        ? ref oneVectorAwayFromEnd
+                        : ref currentSearchSpace;
+
+                    Vector256<short> source0 = Vector256.LoadUnsafe(ref firstVector);
+                    Vector256<short> source1 = Vector256.LoadUnsafe(ref oneVectorAwayFromEnd);
+
+                    Vector256<byte> result = IndexOfAnyLookup<TNegator, TOptimizations>(source0, source1, asciiBitmap256, secondBitmap256, offset256);
+                    if (result != Vector256<byte>.Zero)
+                    {
+                        return TResultMapper.FirstIndexOverlapped<TNegator>(ref searchSpace, ref firstVector, ref oneVectorAwayFromEnd, result);
+                    }
+                }
+
+                return TResultMapper.NotFound;
+            }
+
+            Vector128<byte> asciiBitmap = state.AsciiBitmap128();
+            Vector128<byte> secondBitmap = state.SecondBitmap128();
+            Vector128<ushort> offset = Vector128.Create(state.Offset);
+
+#pragma warning disable IntrinsicsInSystemPrivateCoreLibAttributeNotSpecificEnough // The behavior of the rest of the function remains the same if Avx2.IsSupported is false
+            if (!Avx2.IsSupported && searchSpaceLength > 2 * Vector128<short>.Count)
```

Comment:
Why is the !Avx2.IsSupported check needed here?

Reply:
Logically, the branches here are set up something like
```c#
if (searchSpaceLength > 16)
{
    if (Avx2.IsSupported)
    {
        if (searchSpaceLength > 32)
        {
            // Process the input with 2 Vector256s (32 chars) at a time.
        }

        // Process the last 1-32 chars with 2 overlapped Vector256s.
        return NotFound;
    }
    else
    {
        // Process the input with 2 Vector128s (16 chars) at a time.
        // Need Vector128 state here
    }
}

// Process the last 1-16 chars with 2 overlapped Vector128s.
// Need Vector128 state here
return NotFound;
```
But note that we'd need the `Vector128` state in both the nested `else`, and the final block at the end.
That means we'd either have to
- load it twice, or
- load it at the very beginning, but that's just overheard for the `Vector256` path that will never need it

---

To avoid that, the branches are instead written like
```c#
if (Avx2.IsSupported && searchSpaceLength > 16)
{
    // Load Vector256 state
    if (searchSpaceLength > 32)
    {
        // Process the input with 2 Vector256s (32 chars) at a time.
    }

    // Process the last 1-32 chars with 2 overlapped Vector256s.
    return NotFound;
}

// Load Vector128 state
if (!Avx2.IsSupported && searchSpaceLength > 16)
{
    // Process the input with 2 Vector128s (16 chars) at a time.
}

// Process the last 1-16 chars with 2 overlapped Vector128s.
return NotFound;
```

The `!Avx2.IsSupported` check is there because we've already handled longer inputs if we had `Avx2` support.

Reply:
Ok, so this is really:
```C#
if (searchSpaceLength > 16)
{
    if (Avx2.IsSupported)
    {
        ...
    }
    else
    {
        ...
    }
}
```